### ISSUE TITLE
Rewrite in ES7

### ISSUE THREAD

**Issue by cybertk** [cybertk/abao#223]
As ES7 brings most of Coffee Script features, it's time to roll it.

**Comment by oshalygin** [comment]
Let's get started gents, strategy for this?  I'm proposing a feature branch as we start committing against it and making sure things are working.  That gives us the ability to keep the master branch clean in case there are critical updates we have to make and still be able to push to npm.  

pruposed branch: `feature/es7-migration`.

**Comment by cybertk** [comment]
Good to go

**Comment by oshalygin** [comment]
I'll push the branch and start some of the initial work.  There are a couple of things to figure out, I'll create issues for each one for discussion so that we can move efficiently.

**Comment by sichvoge** [comment]
@oshalygin should we add a milestone for that so that we can see the progress?

**Comment by oshalygin** [comment]
@sichvoge yeah lets do it.  I've been on vacation in Seattle for two weeks and just got back, I'm going to get rolling again on this project.

**Comment by sichvoge** [comment]
Cool, btw - we should probably sit down at some point and talk about the current parser and what we (RAML Workgroup) are doing around the [API Modeling Framework](http://forums.raml.org/t/new-api-modeling-framework-now-available-as-open-source/1819?u=christian_vogel).

In short, we will replace our current parsers with that as it provides a more formal description of the actual parsed model with well defined programmatic interfaces. It will simplify a lot.

What that actually means, we could talk over Hangout or Zoom. What do you think?

It should still not stop us to go the first step on migrating the current implementation to ES7.

**Comment by oshalygin** [comment]
I'm down.  Do we want to start this over Gitter/Slack?  Sorry I've been bad about responding

**Comment by sichvoge** [comment]
I can add you to a Slack if you like. I am back in action on Monday as I am currently relocating!

**Comment by oshalygin** [comment]
Awesome !!  Don't break your back 📦

**Comment by sichvoge** [comment]
Can you guys send me an email to christian (dot) vogel (at) mulesoft (dot) com. Need your email address to add you.

**Comment by shumsky** [comment]
Hi guys,

Is this issue still relevant? Does the branch `es7-migration` contain the latest work on this?

**Comment by plroebuck** [comment]
@shumsky Sorry for late reply. Had nothing to do with this branch, but from what I can tell it looks like it was branched and the README modified -- so I'd say not...
